### DISCUSSION TITLE
Self-hosted github runner in docker on macos

### DISCUSSION THREAD

**Discussion by bldrvnlw** [community/community#36570] [Other Feature Feedback, Questions, & Ideas]
We need a self-hosted github runner that can build C++ using XCode, to reduce GitHub costs (currently using mac machines on github). 

We have hardware (a mac-mini x86-46)  with macOS 12.6 (Monterey) and the necessary development software but would prefer to run the MacOS self-hosted runner (actions-runner-osx-x64-2.298.2.tar.gz)  in a docker container on the mac to provide more isolation from our internal network.

Is this even possible? If so are there examples of how to do it?

**Comment by AF250329** [comment]
Hi

In your repository you should click on `Settings` on top menu. Then click on `Actions` on the left menu and click on `Runners` in drop down menu like this:

![image](https://user-images.githubusercontent.com/78792938/196535970-6b5a8dab-2d06-4e8a-a09b-b4cd76df1dcb.png)

At `Runners` page you will see the `New self-hosted runner` button on the top-right. Just click on this button and you will be navigated with detailed page on how to setup self-hosted runner in any OS (including Mac OS)

![image](https://user-images.githubusercontent.com/78792938/196536321-ab9511df-90eb-48a2-b00a-d08f662e0f11.png)

Try to follow this instruction to set up runner in Mac OS or Docker if you need to

**Reply by bldrvnlw** [reply]
The issue, if I understand it correctly, is that there needs to be a docker base MacOS image for to install the MacOS self-hosted runner. This is not supported by Apple yet. Linux (open source) if of course possible and Microsoft have added Windows container support but not MacOS yet.

**Reply by AF250329** [reply]
I don't really understand (may be because I don't work with Mac OS ?) 

Here is end-to-end tutorial from Microsoft on how to create and run .NET Core app inside Docker image on Mac OS: [Get started with Docker in Visual Studio for Mac](https://learn.microsoft.com/en-us/visualstudio/mac/docker-quickstart?view=vsmac-2022) - so Mac OS can run Docker ? Even according to Docker itself: [Install Docker Desktop on Mac](https://docs.docker.com/desktop/install/mac-install/)

So once Docker installed you need to pull image that has Powershell installed and then just run command from 'instructions' page on how to install self-hosted runner. The very first command is to download actual 'self-runner' application. 

If you can not find image that contains Powershell just run something else - `bash` may be ?

**Comment by bldrvnlw** [comment]
So far I've come across two possibilities:

1. SickCodes Docker-OSX https://github.com/sickcodes/Docker-OSX
2. MacOS containers in containerd, https://github.com/containerd/containerd/discussions/5525, though this is a work in progress

**Reply by bldrvnlw** [reply]
There are some questions about GitHub runners on the SickCodes Discord related to Docker-OSX

**Comment by ossanna16** [comment]
Hey there @bldrvnlw and thank you for asking a good question! I'm glad you found our community 🙂
